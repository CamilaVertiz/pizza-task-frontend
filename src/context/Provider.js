import React, { useEffect } from 'react'
import { useLocalStorage } from '../utils/useLocalStorage'

export const InitialState = {
    currency: { type: 'usd', conversion: 1, shipping: 0 },
    shippingFixed: 6.50,
    cart: [],
    header: true,
}

export const Context = React.createContext();
    const Provider = (props) => {
            const [state, setState] = useLocalStorage('globalKey', InitialState)

        const setCart = (cart) => {
            setState({
            ...state,
            cart,
            })
        }

        const setCurrency = (currency) => {
            setState({
            ...state,
            currency,
        })}

        const setHeader = (header) => {
            setState({
            ...state,
            header,
        })}
        
        let quantity = state.cart.reduce(function(prev, current) {
            return prev + current.qty
        }, 0);

        let total = (state.cart.reduce(function(prev, current) {
            return prev + (+current.price * current.qty) 
        }, state.cart.length > 0 ? state.shippingFixed : 0) * state.currency.conversion).toFixed(2);

        useEffect(() => {
            let shipping = 0;
            if( state.cart.length > 0)
                shipping = (state.shippingFixed * state.currency.conversion).toFixed(2);

            setCurrency({ ...state.currency, shipping})
        }, [state.cart]);    
        
        return (
            <Context.Provider value={{ state, setCurrency, setCart, setHeader, total, quantity}}>
            {props.children}
            </Context.Provider>
        )
    }

export default Provider;