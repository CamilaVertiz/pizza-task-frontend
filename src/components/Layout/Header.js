import React, { useContext } from 'react'
import { Context } from '../../context/Provider'
import { Link } from 'react-router-dom'
import Cart from '../Cart/Cart'
import axios from 'axios'

const Header = () => {
    const { state, setCurrency, total, quantity } = useContext(Context)

    const changeCurrency = async () => {
        if(state.currency.type === 'usd'){
             try{       
                 const conRes = await axios.get('https://api.exchangeratesapi.io/latest?base=USD') 
                 setCurrency({ ...state.currency, type: 'eur', conversion: conRes.data.rates.EUR , shipping: (state.shippingFixed * conRes.data.rates.EUR).toFixed(2) })            
             } catch(err){
                console.log(err)
             }
        }else{
             setCurrency({ ...state.currency, type:'usd', conversion: 1, shipping: state.shippingFixed * 1 })
        }       
    }
    return (
        <div className="header">
            <header className="header-fixed">
                <div className="header-limiter">
                    <a href="/"><span className="logo"></span> <h1 className="title">Pizza Task</h1></a> 
                   
                    <nav>
                        <h1>
                            <span className="currency-exchange" onClick={changeCurrency}></span> 
                            <span className="total"><i className={`fa fa-${state.currency.type}`} aria-hidden="true"/>{ total }</span>
                            <span>
                                <span className="dropdown">
                                    <i className="fa fa-shopping-cart"></i>                                
                                    <span className="cart-count">{ quantity }</span>
                                    <div className={`dropdown-content ${state.cart.length > 0 ? "" : "hidden"}`}>
                                        <Cart />
                                        <Link to="/checkout" className="link">checkout  <i className="fa fa-arrow-right" aria-hidden="true"></i></Link>
                                    </div>
                                </span>     
                            </span>
                        </h1>
                    </nav>
                </div>
            </header>

            <div className="header-fixed-placeholder"></div>
        </div>
   );
}

export default Header;