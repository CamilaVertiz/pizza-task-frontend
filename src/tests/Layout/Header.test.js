import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../components/Layout/Header';
import { Context, InitialState } from '../../context/Provider'
import { BrowserRouter as Router } from 'react-router-dom'

describe('Header component', () => {
    const state = { ...InitialState };
    const setCurrency = jest.fn();
    const total = 0;
    const quantity = 0;
  it('Snapshot', () => {
    const tree = renderer
    .create(<Context.Provider value={{ state, setCurrency, total, quantity }}><Router><Header/></Router></Context.Provider>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
});
