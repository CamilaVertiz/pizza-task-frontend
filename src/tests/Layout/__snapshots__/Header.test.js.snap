// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Snapshot 1`] = `
<div
  className="header"
>
  <header
    className="header-fixed"
  >
    <div
      className="header-limiter"
    >
      <a
        href="/"
      >
        <span
          className="logo"
        />
         
        <h1
          className="title"
        >
          Pizza Task
        </h1>
      </a>
      <nav>
        <h1>
          <span
            className="currency-exchange"
            onClick={[Function]}
          />
          <span
            className="total"
          >
            <i
              aria-hidden="true"
              className="fa fa-usd"
            />
            0
          </span>
          <span>
            <span
              className="dropdown"
            >
              <i
                className="fa fa-shopping-cart"
              />
              <span
                className="cart-count"
              >
                0
              </span>
              <div
                className="dropdown-content hidden"
              >
                <ul
                  className="list-unstyled"
                >
                  <li
                    className="text-right"
                  >
                    <small>
                      Shipping 
                      <i
                        aria-hidden="true"
                        className="fa fa-usd"
                      />
                       
                      0
                       
                    </small>
                  </li>
                  <li
                    className="text-right"
                  >
                    <span
                      className="font-weight-bold"
                    >
                      Total 
                      <i
                        aria-hidden="true"
                        className="fa fa-usd"
                      />
                      0
                       
                    </span>
                  </li>
                </ul>
                <a
                  className="link"
                  href="/checkout"
                  onClick={[Function]}
                >
                  checkout  
                  <i
                    aria-hidden="true"
                    className="fa fa-arrow-right"
                  />
                </a>
              </div>
            </span>
          </span>
        </h1>
      </nav>
    </div>
  </header>
  <div
    className="header-fixed-placeholder"
  />
</div>
`;
